beekeeper-metadata-cleanup/src/test/java/com/expediagroup/beekeeper/metadata/cleanup/handler/GenericMetadataHandlerTest.java

```diff
@@ -0,0 +1,255 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.metadata.cleanup.handler;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import static com.expediagroup.beekeeper.core.model.HousekeepingStatus.DELETED;
+import static com.expediagroup.beekeeper.core.model.HousekeepingStatus.FAILED;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+
+import com.expediagroup.beekeeper.cleanup.aws.S3PathCleaner;
+import com.expediagroup.beekeeper.cleanup.hive.HiveMetadataCleaner;
+import com.expediagroup.beekeeper.core.model.HousekeepingMetadata;
+import com.expediagroup.beekeeper.core.repository.HousekeepingMetadataRepository;
+
+@ExtendWith(MockitoExtension.class)
+public class GenericMetadataHandlerTest {
+
+  private @Mock HousekeepingMetadataRepository housekeepingMetadataRepository;
+  private @Mock S3PathCleaner pathCleaner;
+  private @Mock HiveMetadataCleaner metadataCleaner;
+  private @Mock HousekeepingMetadata mockMetadata;
+  private @Mock Pageable mockPageable;
+  private @Mock Pageable nextPageable;
+  private @Mock PageImpl<HousekeepingMetadata> mockPage;
+
+  private ExpiredMetadataHandler handler;
+
+  private static final String DATABASE = "database";
+  private static final String TABLE_NAME = "tableName";
+  private static final String PARTITION_NAME = "event_date=2020-01-01/event_hour=0/event_type=A";
+  private static final LocalDateTime CLEANUP_INSTANCE = LocalDateTime.ofInstant(Instant.now(), ZoneOffset.UTC);
+
+  @BeforeEach
+  public void init() {
+    when(mockPage.getContent()).thenReturn(List.of(mockMetadata));
+    when(mockMetadata.getDatabaseName()).thenReturn(DATABASE);
+    when(mockMetadata.getTableName()).thenReturn(TABLE_NAME);
+    when(mockMetadata.getPartitionName()).thenReturn(null);
+
+    handler = new ExpiredMetadataHandler(housekeepingMetadataRepository, metadataCleaner, pathCleaner);
```

Comment:
So you're actually testing an implementation. I would remove this class and move all the tests to ExpiredMetadataHandlerTest as that is the only implementation you're actually need to test.